features/upgrade/asb/upgrade.feature

```diff
@@ -0,0 +1,114 @@
+    #   | f | <%= ENV['BUSHSLICER_HOME'] %>/testdata/olm/art-secret-ansible.yaml |
+    #   | p | NAME=aosqe4-secret                                                                              |
+    #   | p | TOKEN=<your quay token>                                                                         |
+    # Then the step should succeed
+    # When I process and create:
+    #   | f | <%= ENV['BUSHSLICER_HOME'] %>/testdata/olm/operatorsource-ansible.yaml |
+    #   | p | NAME=aosqe4-operators                                                                               |
+    #   | p | SECRET=aosqe4-secret                                                                                |
+    #   | p | REGISTRY=aosqe4                                                                                     |
+    # Then the step should succeed
+    # Enable Service Catalog so that the ASB can work well
+    Given enable service catalog
+    # Get the cluster version: oc get clusterversion version -o=jsonpath='{.spec.channel}' | tr -d 'stable-'
+    Given the major.minor version of the cluster is stored in the clipboard
+
+    # This step will install the OperatorGroup, operator. Please use your operator Package name in here.
+    Given optional operator "openshiftansibleservicebroker" from channel "<%= cb.operator_channel_name %>" is subscribed in "openshift-ansible-service-broker" project
+    # Check the pods of the operator
+    Then I wait for the "openshift-ansible-service-broker-operator" deployment to appear
+    And I wait up to 180 seconds for the steps to pass:
+    """
+    And a pod becomes ready with labels:
+      | name=openshift-ansible-service-broker-operator-alm-owned |
+    """
+    # Check if the corresponding CRD is ready
+    Then I run the :get client command with:
+      | resource | automationbroker |
+    And the output should contain "No resources found"
+    #Create Cluster Role Binding for the operator
+    When I process and create:
+      | f | <%= ENV['BUSHSLICER_HOME'] %>/testdata/olm/asb-rolebinding-template.yaml |
+    Then the step should succeed
+    # Create customer resource for the operator
+    When I process and create:
+      | f | <%= ENV['BUSHSLICER_HOME'] %>/testdata/olm/asb-cr-template.yaml |
+    Then the step should succeed
+    And I wait up to 180 seconds for the steps to pass:
+    """
+    And a pod becomes ready with labels:
+        | app=ansible-service-broker |
+    """
+    
+  @admin
+  @upgrade-check
+  @users=upuser1,upuser2
+  Scenario: upgrade ASB
+    Given I switch to cluster admin pseudo user
+    And I use the "openshift-ansible-service-broker" project
+    # Check if the previous ASB works works well.
+    When I run the :describe client command with:
+      | resource           | clusterservicebroker    |
+      | name               | ansible-service-broker |
+    And the output should match:
+      | Reason:\\s+FetchedCatalog |
+      | Status:\\s+True           |
```

Comment:
This is a very vague check. There might be multiple places with `Status:\\s+True`. So condition (if I understand correctly) `Reason:\\s+FetchedCatalog` might be `Status: False` while something else be `Status: true` and check gets confused. That's we better remove `describe` step and use:
```
  Given admin waits for the "ansible-service-broker" cluster_service_broker to become ready
```